pkg/nuctl/command/patch.go

```diff
@@ -0,0 +1,407 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/dashboard/resource"
+	"github.com/nuclio/nuclio/pkg/errgroup"
+	"github.com/nuclio/nuclio/pkg/platform"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/nuclio/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	FunctionsEndpoint     = "functions"
+	DefaultRequestTimeout = "60s"
+	DefaultConcurrency    = 10
+	StateSuccess          = "success"
+	StateFailed           = "failed"
+	StateSkipped          = "skipped"
```

Comment:
why public and what are those states represent? keep the convention and use `type` if needed to provide clear idea of what its kind and what it represent